src/tsolvers/egraph/UFInterpolator.cc

```diff
@@ -295,6 +295,42 @@ void UFInterpolator::colorCongruenceEdge(CEdge * edge) {
     }
 }
 
+icolor_t UFInterpolator::determineDisequalityColor(PTRef t1, PTRef t2, std::map<PTRef, icolor_t> const & conflictColors) const {
+    icolor_t conf_color = icolor_t::I_UNDEF;
+    PTRef eq = logic.mkEq(t1, t2);
+    if (conflictColors.find(eq) != conflictColors.end()) {
+        conf_color = conflictColors.at(eq);
+        if (conf_color == icolor_t::I_AB) {
+            conf_color = resolveABColor();
+        }
+    } else {
+        auto it = std::find_if(conflictColors.begin(), conflictColors.end(), [this](auto const & entry) { return logic.isDisequality(entry.first); });
+        if (it != conflictColors.end()) {
+            Pterm const & distinctTerm = logic.getPterm(it->first);
+            bool t1Present = std::find(distinctTerm.begin(), distinctTerm.end(), t1) != distinctTerm.end();
+            bool t2Present = std::find(distinctTerm.begin(), distinctTerm.end(), t2) != distinctTerm.end();
+            assert(t1Present and t2Present);
+            if (not(t1Present and t2Present)) {
+                throw OsmtInternalException("Error in UF interpolator, could not determine the color of the conflict equality");
+            } else {
```

Comment:
This is essentially debug code but it could have an impact on performance, right?  Would it make sense to extract it as a lambda and put inside an assert?

Reply:
The performance shouldn't be an issue.  The remaining question would be whether putting the lambda and its invocation inside the if statement.  Whether this is more clear seems to be a matter of taste.